### PR TITLE
[specific ci=11-03-Upgrade-InsecureRegistry] restore test 11-03-upgrade-insecureRegistry for testing

### LINKED ISSUE #5473: Insecure registry test failure: FetchAuthToken
Failed these jobs:
https://ci.vcna.io/vmware/vic/11504
https://ci.vcna.io/vmware/vic/11495

```
Jun 13 2017 22:20:28.540Z DEBUG Trying https scheme for &imagec.Options{Reference:(*reference.taggedRef)(0xc42060ff00), Registry:"192.168.31.236", Image:"test/busybox", Tag:"latest", Destination:"/tmp", Host:"127.0.0.1:2377", Storename:"564dd154-83f0-7b14-bd03-3e9e2ab1a7a4", Username:"<user>", Password:"<password>", Token:(*fetcher.Token)(nil), Timeout:3600000000000, Outstream:(*ioutils.WriteFlusher)(0xc420180550), InsecureSkipVerify:false, InsecureAllowHTTP:true, ImageManifestSchema1:(*imagec.Manifest)(nil), ImageManifestSchema2:(*schema2.DeserializedManifest)(nil), ManifestDigest:"", RegistryCAs:(*x509.CertPool)(0xc42045e3f0)}
Jun 13 2017 22:20:28.540Z DEBUG URL: https://192.168.31.236/v2/
Jun 13 2017 22:20:28.540Z DEBUG [BEGIN] [github.com/vmware/vic/pkg/fetcher.(*URLFetcher).Head:361] https://192.168.31.236/v2/
Jun 13 2017 22:20:28.586Z DEBUG [ END ] [github.com/vmware/vic/pkg/fetcher.(*URLFetcher).Head:361] [45.84525ms] https://192.168.31.236/v2/
Jun 13 2017 22:20:28.588Z DEBUG [ END ] [github.com/vmware/vic/lib/imagec.LearnRegistryURL:75] [47.30936ms] 192.168.31.236
Jun 13 2017 22:20:28.589Z DEBUG [BEGIN] [github.com/vmware/vic/lib/imagec.LearnAuthURL:100] 192.168.31.236/test/busybox:latest
Jun 13 2017 22:20:28.589Z DEBUG URL: https://192.168.31.236/v2/test/busybox/manifests/latest
Jun 13 2017 22:20:28.589Z DEBUG [BEGIN] [github.com/vmware/vic/pkg/fetcher.(*URLFetcher).Head:361] https://192.168.31.236/v2/test/busybox/manifests/latest
Jun 13 2017 22:20:28.613Z DEBUG [ END ] [github.com/vmware/vic/pkg/fetcher.(*URLFetcher).Head:361] [23.910073ms] https://192.168.31.236/v2/test/busybox/manifests/latest
Jun 13 2017 22:20:28.613Z DEBUG [ END ] [github.com/vmware/vic/lib/imagec.LearnAuthURL:100] [23.973389ms] 192.168.31.236/test/busybox:latest
Jun 13 2017 22:20:28.613Z DEBUG [BEGIN] [github.com/vmware/vic/lib/imagec.FetchToken:144] https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service
Jun 13 2017 22:20:28.613Z DEBUG URL: https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service
Jun 13 2017 22:20:28.613Z DEBUG [BEGIN] [github.com/vmware/vic/pkg/fetcher.(*URLFetcher).FetchAuthToken:198] https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service
Jun 13 2017 22:20:28.613Z DEBUG [BEGIN] [github.com/vmware/vic/pkg/fetcher.(*URLFetcher).Fetch:124] https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service
Jun 13 2017 22:20:28.613Z DEBUG Setting BasicAuth: <user>
Jun 13 2017 22:20:28.613Z DEBUG Setting user-agent to vic/v1.1.0-rc3
Jun 13 2017 22:20:28.626Z ERROR Fetch (https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service) to string error: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: connection refused
Jun 13 2017 22:20:28.626Z DEBUG Download failed, retrying: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: connection refused
Jun 13 2017 22:20:33.626Z DEBUG Setting BasicAuth: <user>
Jun 13 2017 22:20:33.626Z DEBUG Setting user-agent to vic/v1.1.0-rc3
Jun 13 2017 22:20:33.628Z ERROR Fetch (https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service) to string error: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: connection refused
Jun 13 2017 22:20:33.628Z DEBUG Download failed, retrying: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: connection refused
Jun 13 2017 22:20:43.629Z DEBUG Setting BasicAuth: <user>
Jun 13 2017 22:20:43.629Z DEBUG Setting user-agent to vic/v1.1.0-rc3
Jun 13 2017 22:20:43.633Z ERROR Fetch (https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service) to string error: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: connection refused
Jun 13 2017 22:20:43.633Z DEBUG Download failed, retrying: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: connection refused
Jun 13 2017 22:20:58.633Z DEBUG Setting BasicAuth: <user>
Jun 13 2017 22:20:58.633Z DEBUG Setting user-agent to vic/v1.1.0-rc3
Jun 13 2017 22:21:16.670Z ERROR Fetch (https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service) to string error: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: no route to host
Jun 13 2017 22:21:16.670Z DEBUG Download failed, retrying: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: no route to host
Jun 13 2017 22:21:36.670Z DEBUG Setting BasicAuth: <user>
Jun 13 2017 22:21:36.670Z DEBUG Setting user-agent to vic/v1.1.0-rc3
Jun 13 2017 22:21:39.670Z ERROR Fetch (https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service) to string error: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: no route to host
Jun 13 2017 22:21:39.670Z DEBUG Hit max download attempts. Download failed: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: no route to host
Jun 13 2017 22:21:39.670Z DEBUG [ END ] [github.com/vmware/vic/pkg/fetcher.(*URLFetcher).Fetch:124] [1m11.056943456s] https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service
Jun 13 2017 22:21:39.670Z ERROR Download failed: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: no route to host
Jun 13 2017 22:21:39.670Z DEBUG [ END ] [github.com/vmware/vic/pkg/fetcher.(*URLFetcher).FetchAuthToken:198] [1m11.057005817s] https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service
Jun 13 2017 22:21:39.670Z ERROR FetchToken (https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service) failed: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: no route to host
Jun 13 2017 22:21:39.670Z DEBUG [ END ] [github.com/vmware/vic/lib/imagec.FetchToken:144] [1m11.05707475s] https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service
Jun 13 2017 22:21:39.670Z ERROR Failed to fetch OAuth token: FetchToken (https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service) failed: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: no route to host
Jun 13 2017 22:21:39.670Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Image).PullImage:325] [1m11.143906612s] 
Jun 13 2017 22:21:39.670Z ERROR Handler for POST /v1.23/images/create returned error: FetchToken (https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service) failed: Get https://localhost.localdom.ci.vmware.run/service/token?scope=repository%3Atest%2Fbusybox%3Apull&service=token-service: dial tcp 192.168.31.160:443: getsockopt: no route to host
```

So, for some reason it looks like midway through this process we switched from the valid IP address of 192.168.31.236  to 192.168.31.160.  On a different successful run, it did not switch IPs.

### PR DESCRIPTION
**Pull request by chengwang86**
Fixes #5473 

We decided to restore this test because I couldn't reproduce it on either local test env or full ci.

In order to avoid dangling harbor instances on the test server to affect this test, I use `cleanup Harbor` keyword in this PR to clean up these dangling harbor instances before we deploy a new harbor.

### CODE REVIEW

File: tests/test-cases/Group11-Upgrade/11-03-Upgrade-InsecureRegistry.robot [hunk 1]
```diff
@@ -90,40 +90,37 @@ Cleanup Test Environment
...
-    #Install VIC with version to Test Server  ${test_vic_version}  --insecure-registry ${harbor_ip} --no-tls
+    Install VIC with version to Test Server  ${test_vic_version}  --insecure-registry ${harbor_ip} --no-tls
 
-    #Test VCH And Registry  %{VCH-IP}:%{VCH-PORT}  ${harbor_ip}
+    Test VCH And Registry  %{VCH-IP}:%{VCH-PORT}  ${harbor_ip}
 
-    #Upgrade
-    #Check Upgraded Version
-    #Test VCH And Registry  %{VCH-IP}:%{VCH-PORT}  ${harbor_ip}
+    Upgrade
```
Comment (mdharamadas1) [inline tests/test-cases/Group11-Upgrade/11-03-Upgrade-InsecureRegistry.robot]: This looks like a comment and not a keyword?

Comment (chengwang86) [inline tests/test-cases/Group11-Upgrade/11-03-Upgrade-InsecureRegistry.robot]: It is a keyword defined in VCH-Util.robot

File: tests/test-cases/Group11-Upgrade/11-03-Upgrade-InsecureRegistry.robot [hunk 2]
```diff
@@ -90,40 +90,41 @@ Cleanup Test Environment
...
+    Set Test Variable  ${harbor_name}  ${https_harbor_name}
+    ${ip}=  Install Harbor To Test Server  ${harbor_name}  https
+    Set Test Variable  ${harbor_ip}  ${ip}
+    Add Project On Registry  ${harbor_ip}  https
+    ${hdl}  ${pid}=  Setup Test Environment  ${harbor_ip}
+    Set Test Variable  ${handle}  ${hdl}
+    Set Test Variable  ${docker_daemon_pid}  ${pid}
+
+    ${harbor_cert}=  Fetch Harbor Self Signed Cert  ${harbor_ip}
+    Install VIC with version to Test Server  ${test_vic_version}  --insecure-registry ${harbor_ip} --no-tls --registry-ca ${harbor_cert}
```
Comment (gigawhitlocks) [inline tests/test-cases/Group11-Upgrade/11-03-Upgrade-InsecureRegistry.robot]: why `--no-tls` ?

Comment (chengwang86) [inline tests/test-cases/Group11-Upgrade/11-03-Upgrade-InsecureRegistry.robot]: This is in the original test file. I just un-commented it.   I think we can leave it as it is now, just to make sure our ci test is stable. Then we can improve those tests gradually, maybe after this release.

**Pull request merged**